server/src/main/java/org/opensearch/common/geo/PointParser.java

```diff
@@ -0,0 +1,227 @@
+    }
+
+    private Point parseGeoJsonFields(final XContentParser parser, final boolean ignoreZValue) throws IOException {
+        boolean isTypePoint = false;
+        Point point = null;
+        for (int i = 0; i < 2; i++) {
+            if (parser.currentToken() != XContentParser.Token.FIELD_NAME) {
+                throw new OpenSearchParseException(ERR_MSG_INVALID_TOKEN, parser.currentToken());
+            }
+
+            String field = parser.currentName();
+            parser.nextToken();
+            if (FN_GEOJSON_TYPE.equals(field)) {
+                if (parser.currentToken() != XContentParser.Token.VALUE_STRING) {
+                    throw new OpenSearchParseException("{} must be a string", FN_GEOJSON_TYPE);
+                }
+
+                if (!FV_GEOJSON_TYPE_POINT.equals(parser.text())) {
+                    throw new OpenSearchParseException("{} must be {}", FN_GEOJSON_TYPE, FV_GEOJSON_TYPE_POINT);
+                }
+                isTypePoint = true;
+            } else if (FN_GEOJSON_COORDS.equals(field)) {
+                if (parser.currentToken() != XContentParser.Token.START_ARRAY) {
+                    throw new OpenSearchParseException("{} must be an array", FN_GEOJSON_COORDS);
+                }
+                point = parseFromArray(parser, ignoreZValue, FN_GEOJSON_COORDS);
+            } else {
+                throw new OpenSearchParseException(invalidFieldErrMsg);
+            }
+            parser.nextToken();
+        }
+
+        if (!isTypePoint) {
+            throw new OpenSearchParseException("field [{}] missing", FN_GEOJSON_TYPE);
+        }
+
+        if (point == null) {
+            throw new OpenSearchParseException("field [{}] missing", FN_GEOJSON_COORDS);
+        }
+
+        return point;
+    }
+
+    public Point parseFromArray(final XContentParser parser, final boolean ignoreZValue, final String fieldName) throws IOException {
+        double x = Double.NaN;
+        double y = Double.NaN;
+
+        if (parser.currentToken() == XContentParser.Token.START_ARRAY) {
+            parser.nextToken();
+        }
+
+        int element = 0;
+        while (parser.currentToken() != XContentParser.Token.END_ARRAY) {
+            if (parser.currentToken() == XContentParser.Token.VALUE_NUMBER) {
+                element++;
+                if (element == 1) {
+                    x = parser.doubleValue();
+                } else if (element == 2) {
+                    y = parser.doubleValue();
+                } else if (element == 3) {
+                    GeoPoint.assertZValue(ignoreZValue, parser.doubleValue());
+                } else {
+                    throw new OpenSearchParseException("[{}] field type does not accept > 3 dimensions", fieldName);
+                }
+            } else {
+                throw new OpenSearchParseException("numeric value expected");
+            }
+            parser.nextToken();
+        }
+        if (element < 2) {
```

Comment:
```suggestion
        if (y == Double.NaN) {
```

Reply:
It does not cover the case when x is not set.